04-cond.Rmd

```diff
@@ -212,6 +212,38 @@ In this case, "either" means "either or both", not "either one or the other but
 > plot_dist(dat[1:5, 10], threshold = 10)                    # samples (rows) 1-5 on day (column) 10
 > ```
 
+
+> ## Challenge - Putting together data analysis, conditionals, and plotting   {.challenge}
+>   + For this challenge, fill in the blanks of the function hist_max() below to create a function that does the following:  
+  
+> ```{r, eval=FALSE}
+1.For each patient (row) in a given file, find the max inflammation value and store these in a variable called 'max_patient_inflammation'.  
+2.Plot the max values into a simple histogram using the function hist() if and only if the spread (difference between largest and smallest) of the max values for all the patients is greater than your desired spread value, otherwise, return the values.  
+```
+
+> ``` 
+> hist_max <- function(dat, spread) {
+>  max_patient_inflammation <- ______(dat, __, max)
+>  difference <- ___(max_patient_inflammation) - ___(max_patient _inflammation)
+>  if (______ > ______) {
+>      hist(max_patient_inflammation)
+>  } else {
+>      return(max_patient_inflammation)
+>  }
+>}
+>```
+
+> ``` {r, include=FALSE}
+> hist_max <- function(dat, spread) {
+>  max_patient_inflammation <- apply(dat, 1, max)
```

Comment:
Please pay attention to the indentation. It should always be 2 spaces. In your function, you sometimes indent by 1 space and other times by 4 spaces.